crates/engine/tree/src/tree/mod.rs

```diff
@@ -1731,12 +1727,8 @@ where
         if header.is_none() {
             // Note: it's fine to return the unsealed block because the caller already has
             // the hash
-            header = self
-                .state
-                .tree_state
-                .block_by_hash(hash)
-                // TODO: clone for compatibility. should we return an Arc here?
-                .map(|block| block.header().clone());
+            header =
+                self.state.tree_state.block_header_by_hash(&hash).map(SealedHeader::into_header)
```

Comment:
could we change the return type to also be `SealedHeader` for consistency, then we dont need this map here, and instead do `.map(SealedHeader::new_unhashed)` above